internal/gitinterface/log_test.go

```diff
@@ -102,4 +108,171 @@ func TestGetCommitsBetweenRange(t *testing.T) {
 		})
 		assert.Equal(t, expectedCommits, commits)
 	})
+	t.Run("Get commits from invalid range", func(t *testing.T) {
+		commits, err := GetCommitsBetweenRange(repo, plumbing.ZeroHash, plumbing.ZeroHash)
+		assert.NotNil(t, err)
+		assert.Nil(t, commits)
+	})
+	t.Run("Get commits from non-existent commit", func(t *testing.T) {
+		nonExistentHash := plumbing.NewHash("1234567890123456789012345678901234567890")
+		commits, err := GetCommitsBetweenRange(repo, nonExistentHash, plumbing.ZeroHash)
+		assert.NotNil(t, err)
+		assert.Nil(t, commits)
+	})
+}
+
+func TestGetCommitsBetweenRangeForMergeCommits(t *testing.T) {
```

Comment:
I see why this is a separate test but a part of me wonders about keeping them organized by making this a `t.Run` in the previous one. :thinking:

Reply:
I think I've come around on this one given the difference in setup.